src/ui/pages/SwapForm/SwapForm.tsx

```diff
@@ -519,11 +540,63 @@ export function SwapForm() {
                 chain={chain}
                 transaction={configureTransactionToBeSigned(currentTransaction)}
                 configuration={swapView.store.configuration.getState()}
+                onAllowanceEdit={() => allowanceDialogRef.current?.showModal()}
               />
             </ViewLoadingSuspense>
           );
         }}
-      ></BottomSheetDialog>
+      />
+
+      <BottomSheetDialog
+        ref={allowanceDialogRef}
+        key="allowance"
+        height="min-content"
+        containerStyle={{ display: 'flex', flexDirection: 'column' }}
+        renderWhenOpen={() => {
+          invariant(chain, 'Chain must be defined');
+          invariant(
+            spendPosition?.asset,
+            'Spend position asset must be defined'
+          );
+          invariant(
+            spendPosition?.quantity,
+            'Spend position quantity must be defined'
+          );
+          const value = new BigNumber(allowanceQuantityBase);
+          const positionBalanceCommon = getPositionBalance(spendPosition);
+          return (
+            <ViewLoadingSuspense>
+              <div
+                style={{
+                  ['--surface-background-color' as string]:
+                    'var(--z-index-1-inverted)',
+                  display: 'grid',
+                  gap: 12,
+                  gridTemplateRows: 'auto 1fr',
+                  flexGrow: 1,
+                }}
+              >
+                <CustomAllowanceForm
+                  asset={spendPosition.asset}
+                  chain={chain}
+                  address={address}
+                  balance={positionBalanceCommon}
+                  requestedAllowanceQuantityBase={UNLIMITED_APPROVAL_AMOUNT}
+                  value={value}
+                  onSubmit={(quantity) => {
+                    setAllowanceQuantityBase(quantity.toString());
+                    allowanceDialogRef.current?.close();
+                  }}
+                  footerRenderArea="custom-allowance-footer"
+                />
+                <Spacer height={20} />
+                <RenderArea name="custom-allowance-footer" />
```

Comment:
I don't like that I had to use the `RenderArea` here, alternatives that that I can think of:

* Provide smth like `renderFooter` callback and do this:
```tsx
renderFooter={() => (
  <>
    <Spacer height={20} />
    ...
  </>
)}
```
* ...

Reply:
Just added an extra `Spacer` to the `CustomAllowanceForm`